Reference/google/protobuf/compiler/plugin.pb.swift

```diff
@@ -79,7 +79,17 @@ public struct Google_Protobuf_Compiler_CodeGeneratorRequest: ProtobufGeneratedMe
   public var fileToGenerate: [String] = []
 
   ///   The generator parameter passed on the command-line.
-  public var parameter: String? = nil
+  private var _parameter: String? = nil
+  public var parameter: String {
+    get {return _parameter ?? ""}
+    set {_parameter = newValue}
+  }
+  public var hasParameter: Bool {
+    return _parameter != nil
+  }
+  public mutating func clearParameter() {
+    return _parameter = nil
+  }
```

Comment:
> I haven't given much thought to this, but do you think we have enough versioning capabilities to make this kind of breaking change to our API in the future?

Not yet, no. It's been in the back of my mind, though. If we look at Objective-C and C++, they use a combination of preprocessor directives and runtime logic to verify whether the generated protos match the runtime library they're being compiled/linked against. We're a bit more limited in what we can do in that regard because of the lack of a preprocessor, but I've experimented with some protocol conformance tricks that we can use to create a type that only compiles when a specific version of the library is imported.